lib/server/monalisa-synchronization/MonALISAInterface.js

```diff
@@ -0,0 +1,169 @@
+     * @param {URL|string} [configuration.dataPassesUrl] url to fetch data passes
+     * @param {URL|string} [configuration.dataPassDetailsUrl] url to fetch data pass details
+     * @param {string|string[]|Buffer|Buffer[]|Object[]} [configuration.pfxCertificateBytes] PFX or PKCS12 encoded private key
+     * and certificate chain @see tls.createSecureContext([options])
+     * @param {string} [configuration.certificatePassphrase] passphrase to the certificate
+     * @param {number} [configuration.yearLowerLimit] indicates how old data are accepted,
+     * year of lhc period that given data pass belongs must be greater or equal
+     */
+    constructor({
+        dataPassesUrl,
+        dataPassDetailsUrl,
+        pfxCertificateBytes,
+        certificatePassphrase,
+        yearLowerLimit,
+    } = {}) {
+        this.logger = new Log(this.constructor.name);
+        this.dataPassesUrl = new URL(dataPassesUrl);
+        this.dataPassDetailsUrl = new URL(dataPassDetailsUrl);
+        this.yearLowerLimit = yearLowerLimit;
+
+        this.httpOptions = {
+            headers: {
+                Accept:	'application/json;charset=utf-8',
+                Connection:	'keep-alive',
+            },
+            agent: pfxCertificateBytes && certificatePassphrase
+                ? new https.Agent({ pfx: pfxCertificateBytes, passphrase: certificatePassphrase }) : undefined,
+        };
+    }
+
+    /**
+     * Get data passes with valid names from MonALISA
+     * @return {DataPass[]} data passes
+     */
+    async getDataPasses() {
+        const payload = await this.fetchDataPasses();
+        return this.processDataPassesPayload(payload);
+    }
+
+    /**
+     * Fetch data passes from MonALISA
+     * @return {object<string, Partial<MonalisaDataPass>>} payload
+     */
+    async fetchDataPasses() {
+        const { default: fetch } = await import('node-fetch');
+        const specificURL = new URL(this.dataPassesUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        return fetch(specificURL, this.httpOptions).then((response) => response.json());
+    }
+
+    /**
+     * Adapt data pass
+     * @param {MonALISADataPass} dataPass data pass
+     * @return {DataPass} adapted data pass
+     */
+    monalisaDataPassToEntity(dataPass) {
+        return {
+            name: dataPass.name,
+            outputSize: dataPass.output_size ? Number(dataPass.output_size) : null,
+            reconstructedEventsCount: dataPass.reconstructed_events,
+            description: dataPass.description,
+            lastRunNumber: dataPass.last_run,
+        };
+    }
+
+    /**
+     * Process server response payload to database entites, remove unnecessary data
+     * @param {object<string, Partial<MonalisaDataPass>>} payload raw data acquired from MonALISA: mapping of Data Pass Name to its parameters
+     * @return {DataPass[]} adapted data
+     */
+    processDataPassesPayload(payload) {
```

Comment:
Your function names should be more explicit.
I would rather name it `_extractDataPassesFromMonALISAPayload`

Reply:
renamed